### PR TITLE
filter_lua: add index to arraylength call

### PR DESCRIPTION
**Pull request by braydonk**
When building with `FLB_FILTER_LUA_USE_MPACK`, compilation fails because this `flb_lua_arraylength` call didn't have an index argument. This PR passes the index argument to the function that I believe to be correct, which is `-1` (assuming based on its usage in the following loop to get individual elements).

Didn't make an issue but I can if necessary

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [N/A] Example configuration file for the change
- [N/A] Debug log output from testing the change

- [N/A] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.
- [N/A] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [N/A] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**

- [N/A] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

**Comment by braydonk** [comment]
Not sure whether backporting is necessary here.

**Comment by braydonk** [comment]
@nokute78 requesting review from you since the blame suggests that you added the `index` argument to this function.

**Review by nokute78 — APPROVED** [review]
LGTM.

**Comment by braydonk** [comment]
Thanks @nokute78!

Just waiting on an approval from another maintainer to get this merged.

**Pull request merged**
